src/main/java/org/folio/rest/impl/VoucherNumberImpl.java

```diff
@@ -37,15 +37,43 @@ public void getVoucherStorageVoucherNumber(String lang, Map<String, String> okap
           if(reply.succeeded()) {
             String voucherNumber = reply.result().getList().get(0).toString();
             log.debug("Retrieved voucher number: {}", voucherNumber);
-            asyncResultHandler.handle(succeededFuture(respond200WithApplicationJson(new SequenceNumber().withSequenceNumber(voucherNumber))));
+            SequenceNumber sequenceNumber = new SequenceNumber().withSequenceNumber(voucherNumber);
+            asyncResultHandler.handle(succeededFuture(VoucherStorageVoucherNumber.GetVoucherStorageVoucherNumberResponse.respond200WithApplicationJson(sequenceNumber)));
           } else {
             throw new Exception(reply.cause());
           }
         } catch (Exception e) {
           log.error(e.getMessage(), e);
-          asyncResultHandler.handle(succeededFuture(respond500WithTextPlain(messages.getMessage(lang, MessageConsts.InternalServerError))));
+          String msg = messages.getMessage(lang, MessageConsts.InternalServerError);
+          asyncResultHandler.handle(succeededFuture(VoucherStorageVoucherNumber.GetVoucherStorageVoucherNumberResponse.respond500WithTextPlain(msg)));
         }
       });
     });
   }
+
+  @Override
+  public void postVoucherStorageVoucherNumberStartByValue(String value, String lang, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    vertxContext.runOnContext((Void v) -> {
+      if(NumberUtils.isDigits(value)) {
+        String tenantId = TenantTool.calculateTenantId(okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT));
+        PostgresClient.getInstance(vertxContext.owner(), tenantId).execute(String.format(SET_START_SEQUENCE_VALUE_QUERY, value), reply -> {
+          try {
+            if(reply.succeeded()) {
+              log.debug("(Re)set start value for voucher number sequence: {}", value);
+              asyncResultHandler.handle(succeededFuture(VoucherStorageVoucherNumber.PostVoucherStorageVoucherNumberStartByValueResponse.respond204()));
+            } else {
+              throw new Exception(reply.cause());
```

Comment:
Sonarqube doesn't like this... please throw a specific exception...

Reply:
Wait, can you just throw `reply.cause()`?

Reply:
@craigmcnally, thanks. 
Moreover, DB reply error processing via exception handling looks redundant here since there are no other operations besides verification the DB reply. Therefore, I believe it is sufficient to simply check whether everything is ok with DB reply using if statement.